bundle-workflow/python/test_workflow/local_test_cluster.py

```diff
@@ -51,3 +58,37 @@ def destroy(self):
             self.stdout.close()
             self.stderr.close()
             self.process = None
+
+    def url(self, path=''):
+        return f'{"https" if self.security_enabled else "http"}://{self.endpoint()}:{self.port()}{path}'
+
+    def download(self):
+        print(f'Creating local test cluster in {self.work_dir}')
+        os.chdir(self.work_dir)
+        print(f'Downloading bundle from {self.manifest.build.location}')
+        urllib.request.urlretrieve(self.manifest.build.location, 'bundle.tgz')
+        print(f'Downloaded bundle to {os.path.realpath("bundle.tgz")}')
+
+        print('Unpacking')
+        subprocess.check_call('tar -xzf bundle.tgz', shell = True)
+        print('Unpacked')
+
+    def disable_security(self, dir):
+        subprocess.check_call(f'echo "plugins.security.disabled: true" >> {os.path.join(dir, "config", "opensearch.yml")}', shell = True)
+
+    def wait_for_service(self):
+        print('Waiting for service to become available')
+        url = self.url('/_cluster/health')
+
+        for attempt in range(10):
```

Comment:
Suggestion: [retry](https://pypi.org/project/retry/) could trim this down, and make it easier if these files need more retry functionality in the future

Reply:
Good idea. Future PR!